src/compiler/emitter.ts

```diff
@@ -2687,6 +2687,22 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                 return result;
             }
 
+            function emitEs6ExportDefaultCompat() {
+                if (compilerOptions.module === ModuleKind.CommonJS || compilerOptions.module === ModuleKind.AMD || compilerOptions.module === ModuleKind.UMD) {
+                    if (!hasProperty(currentSourceFile.identifiers, "___esModule")) {
+                        if (languageVersion >= ScriptTarget.ES5) {
+                            // default value of configurable, enumerable, writable are `false`. 
+                            write("Object.defineProperty(exports, \"__esModule\", { value: true });");
+                            writeLine();
+                        }
+                        else {
+                            write("exports.__esModule = true;");
+                            writeLine();

```

Comment:
Move writeLine after the if-else so it only appears once. Not a big deal though.